### PR TITLE
Tests for backwards-compatibility reading/writing/exporting/importing old repo format

### PR DESCRIPTION
**Pull request by cyli**
As a test for #383

### CODE REVIEW

**Comment by riyazdf** [comment]
awesome to have fixtures and docs around this, LGTM!

File: fixtures/compatibility/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	- GUN: `docker.io/notary0.1/samplerepo`
+	- key passwords: "randompass"
+	- targets:
+
+		```
+		  NAME                                DIGEST                                SIZE (BYTES)
+		------------------------------------------------------------------------------------------
+		  v1     454fd6f9e5a71c34902a673f66cc73be5890785b55014afb0eee6cbbe784cca9   243
+		```
+	- timestamp metadata has been removed - we have to rotating the timestamp key in order for the server to be able to sign the timestamp, and the `timestamp.json` generated by the server will be version 1, since as far as it knows this is a new repo.  If we kept the `timestamp.json` with the higher version number, we would not be able to publish.
```
Comment (diogomonica) [inline fixtures/compatibility/README.md]: "we have to rotating the timestamp"

**Comment by diogomonica** [comment]
This is great. Thank you for taking the time to make sure we test the backwards behavior.

LGTM

File: fixtures/compatibility/README.md (hunk 1 above)
Comment (cyli) [inline fixtures/compatibility/README.md]: Oops!  hehe, thanks!  :D

**Pull request merged**